shm/game.cpp

```diff
@@ -0,0 +1,186 @@
+              << "You have got " << money_ << "money in " << days_ << "!\n";
+    printTrail();
+}
+
+bool Game::checkLoseConditions() const {
+    return currentDay_ > days_ || player_->getMoney() == SIZE_MAX;  // change when -= money in player return Size_max
+}
+
+void Game::printLoseScreen() {
+    std::cout << "You have lost, good bye. \n"
+              << "You have failed to get: " << finalGoal_ - money_ << " more money.\n";
+    printTrail();
+}
+
+void Game::printMenu() {
+    std::cout << "Money: " << player_->getMoney() << " | "
+              << "Day: " << time_->getElapsedTime() << " | "
+              << "Days left: " << days_ - time_->getElapsedTime() << " | "
+              << "Money to earn: " << finalGoal_ - money_ << " | ";
+             // << "Current position: " << map_->getCurrentPosition()->getCoordinates() << '\n';
+}
+
+void Game::printOptions() {
+    std::cout << "What do you want to do? \n"
+              << "1. Travel \n"
+              << "2. Buy \n"
+              << "3. Sell \n"
+              << "4. Print cargo \n"
+              << "0. Exit game \n";
+}
+
+void Game::makeAction(Action pickAction) {
+    switch (pickAction) {
+    case Action::exit: {
+        std::cout << "Bye, bye \n";
+        break;
+    }
+    case Action::travel: {
+        travel();
+        break;
+    }
+    case Action::buy: {
+        buy();
+        break;
+    }
+    case Action::sell: {
+        sell();
+        break;
+    }
+    case Action::printCargo: {
+        showCargo();
+        break;
+    }
+    default: {
+        std::cout << "I can't do this! \n";
+    }
+    }
+}
+
+void Game::travel() {
+    while (true) {
+        std::cout << *map_;
+        std::cout << "Where do you want to sail? (posX posY) \n";
+        int posX, posY;
+        std::cin >> posX >> posY;
+        Island* island = map_->getIsland(Coordinates(posX, posY));
+        if (island == map_->getCurrentPosition()) {
+            std::cout << "You are already here! \n";
+        } else if (island != nullptr) {
+            const int travelDistance = map_->getDistanceToIsland(island);
+            const size_t speed = player_->getSpeed();
+            const size_t daysOfTravel = ((travelDistance + speed) / speed);
+            std::cout << "Travel will take " << daysOfTravel << " day/s. Do you want to travel Y/N? ";
+            char travelDecision;
+            std::cin >> travelDecision;
+            if (std::toupper(travelDecision) == 'Y') {
+                map_->travel(island);
+                std::cout << "~~~~" << '\n' << daysOfTravel << " day/s have passed. \n";
+                for (size_t i = 0; i < daysOfTravel; i++) {
+                    ++*time_;
+                }
+                break;
+            }
+        }
+        std::cout << "Give me right coordinates! \n";
+    }
+}
+
+void Game::buy() {
+    Island* currentIsland = map_->getCurrentPosition();
+    std::cout << "\n" << *(currentIsland->getStore().get()) << "\n";
```

Comment:
Hmm I think you dont buy here any product. Only show product available in store.